strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java

```diff
@@ -0,0 +1,113 @@
+package org.carlspring.strongbox.authentication.registry.support;
+
+import org.carlspring.strongbox.authentication.api.Authenticator;
+import org.carlspring.strongbox.authentication.registry.AuthenticatorsRegistry;
+import org.carlspring.strongbox.authentication.registry.support.xml.XmlAuthenticator;
+import org.carlspring.strongbox.authentication.registry.support.xml.XmlAuthenticators;
+import org.carlspring.strongbox.resource.ConfigurationResourceResolver;
+import org.carlspring.strongbox.xml.parsers.GenericParser;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Throwables;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigRegistry;
+import org.springframework.context.annotation.ClassPathBeanDefinitionScanner;
+import org.springframework.core.io.Resource;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+public class AuthenticatorsScanner
```

Comment:
First time I saw your comment I thought it's a great idea. 

Generally, `BeanFactoryPostProcessor` allows me to create/register beans, etc. But I need to provide a class type, at least. I would have to check every class, in the required package, is it a `@Component`, and generally is it a bean, internally. I assume that provided external `Authenticator` may not be a single SpringBean required, as it may use other SpringBeans too. So, in my opinion there is a lot of work internally to lookup what's a bean and register every one properly.

But then I dug into the Spring component scanning internals ... Please take a look at the existing `AuthenticatorsScanner` implementation method called `scanRequiredComponents`:
https://github.com/fuss86/strongbox/blob/SB-625/strongbox-user-management/src/main/java/org/carlspring/strongbox/authentication/registry/support/AuthenticatorsScanner.java#L100
In my opinion, Spring give us all the necessary work for free. I just have to provide an array of packages to scan. Everything is else is done by Spring, under the hood. Who can better understand Spring internals than their authors ?:)

What do you think ?